src/main/java/org/galatea/starter/service/IexService.java

```diff
@@ -45,5 +74,34 @@
     }
   }
 
+  /**
+   * Get the historical price data for a given stock symbol.
+   *
+   * @param symbol stock symbol to get historical data for.
+   * @param range specified time range of the request.
+   * @param date specified date in the format YYYYMMDD.
+   * @return a list of historical price data for the given symbol.
+   */
+  public List<IexHistoricalPrice> getHistoricalPrice(final String symbol, final String range,
+      final String date) {
+    log.info("Retrieving historical price with symbol {}, range {}, date {}", symbol, range, date);
+    boolean isAllLetters = Pattern.matches("[a-zA-Z]+", symbol);
+    if (!symbol.isEmpty() && isAllLetters) {
+      if (date == null) {
+        if (range == null) {
+          return iexCloudClient.getHistoricalPriceForSymbol(symbol);
+        } else if (rangeIsValid(range)) {
+          return iexCloudClient.getHistoricalPriceByRange(symbol, range);
+        }
+      } else if (dateIsValid(date)) {
+        if (range == null) {
+          return iexCloudClient.getHistoricalPriceByDate(symbol, date);
+        } else if (rangeIsValid(range)) {
+          return iexCloudClient.getHistoricalPriceByRangeDate(symbol, range, date);
+        }
+      }
+    }
+    return Collections.emptyList();
```

Comment:
It looks like you return an empty likst for all invalid requests. This is ok, but consider next time returning a 400 (BAD_REQUEST) error and telling the requestor what went wrong.